evap/staff/forms.py

```diff
@@ -181,6 +181,16 @@ def validate_unique(self):
                     name_field = e.params["unique_check"][1]
                     self.add_error(name_field, e)
 
+    def clean_participants(self):
+        participants = self.cleaned_data.get('participants')
+        if self.instance.pk:
+            voters = self.instance.voters.all()
+            removed_voters = [voter.username for voter in voters if voter not in participants]
+            if removed_voters:
+                self.add_error("participants", _("Participants who already voted for the course can't be removed: %s") % ", ".join(removed_voters))
```

Comment:
..but now the user has to manually add that specific person again, right? wouldn't it be nice to save the form and show a warning "these people here haven't been removed"?

Reply:
correct. but i really didn't care ;)
should i re-add the user?

Reply:
since it's probably just one more line, i would say so, yes.

Reply:
shouldn't adding the users to the return value change what's displayed in the form afterwards? i can return a modified list of users, however the removed user stays removed in the form.

Reply:
i feel there's a misunderstanding, i was not suggesting saving the wrong data, but overwriting the user's choice on save, thus saving the correct data, and displaying a warning after the redirect that some of the users choices have been ignored. of course that's not 100% nice, but having to add them manually is not that nice either.

in the end it doesn't really matter as it probably happens once every few years. you make the call.